src/gsy_e/models/market/future.py

```diff
@@ -41,6 +42,39 @@ class FutureMarketException(Exception):
     """Exception specific to the Future markets."""
 
 
+class FutureOrders(UserDict):
+    """Special mapping object to keep track of a future market's orders."""
+    def __init__(self, *args, **kwargs):
+        self.slot_order_mapping = {}
+        super().__init__(*args, **kwargs)
+
+    def __setitem__(self, order_id, order):
+        self.data[order_id] = order
+        if order.time_slot not in self.slot_order_mapping:
+            self.slot_order_mapping[order.time_slot] = []
+        self.slot_order_mapping[order.time_slot].append(order)
+
+    def __delitem__(self, order_id):
+        order = self.data.pop(order_id, None)
+        if order:
+            self.slot_order_mapping[order.time_slot].remove(order)
```

Comment:
Each time we delete an item from this FutureOrders dictionary (`del` or `pop`), this dunder method will keep both mappings in sync